source/slang/slang-ir-autodiff-unzip.h

```diff
@@ -66,6 +86,185 @@ struct DiffUnzipPass
         return unzippedFunc;
     }
 
+    bool isRelevantDifferentialPair(IRType* type)
+    {
+        if (as<IRDifferentialPairType>(type))
+        {
+            return true;
+        }
+        else if (auto argPtrType = as<IRPtrTypeBase>(type))
+        {
+            if (as<IRDifferentialPairType>(argPtrType->getValueType()))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    InstPair splitCall(IRBuilder* primalBuilder, IRBuilder* diffBuilder, IRCall* mixedCall)
+    {
+        IRBuilder globalBuilder;
+        globalBuilder.init(autodiffContext->sharedBuilder);
+
+        auto fwdCallee = as<IRForwardDifferentiate>(mixedCall->getCallee());
```

Comment:
`callee` could have already been generated and therefore in a form of `IRFunc`, or  is there an assumption that this will always happen before generation of the derivative and replacement of the `IRForwardDifferentiate`?

Reply:
This assumption should hold if the function is being handled in reverse-mode (and methods called form this function do not have pre-defined fwd-mode/rev-mode functions) since we transpose the function (and the call inst) before going for the next set of functions to be transcribed.

We do need to additional support here for other cases (like pre-existing user-defined reverse-mode definitions). I'm hoping to add this in the next PR (along with additional testing)